tests/conftest.py

```diff
@@ -86,111 +97,36 @@ def pytest_generate_tests(metafunc):
 
 
 @pytest.fixture(scope="session")
-def minikube(request, worker_id):
-    def teardown():
-        if inst.container and not k8s_skip_teardown:
-            print("Removing %s container ..." % inst.container.name)
-            inst.container.remove(force=True, v=True)
-
-    request.addfinalizer(teardown)
-    k8s_version = os.environ.get("K8S_VERSION")
-    if not k8s_version:
-        k8s_version = request.config.getoption("--k8s-version")
-    k8s_timeout = request.config.getoption("--k8s-timeout")
-    if not k8s_timeout:
-        k8s_timeout = K8S_DEFAULT_TIMEOUT
+def k8s_cluster(request):
+    agent_image_name = request.config.getoption("--agent-image-name")
+
+    kube_context = None
+    if request.config.getoption("--use-minikube"):
+        minikube_container_name = request.config.getoption("--minikube-container-name")
+        dclient = get_docker_client()
+        assert get_docker_client().containers.get(
+            minikube_container_name
+        ), f"You must start the minikube container ({minikube_container_name}) before running pytest by running `make minikube`."
+
+        # Pull the kubeconfig out of the minikube container and make a cluster instance based on it.
+        minikube_cont = dclient.containers.get(minikube_container_name)
+        _, kubeconfig_bytes = minikube_cont.exec_run("cat /kubeconfig")
+
+        _, kube_config_path = tempfile.mkstemp()
+        request.addfinalizer(lambda: os.remove(kube_config_path))
+
+        with open(kube_config_path, "wb") as fd:
```

Comment:
`Path("path").write_bytes(kubeconfig_bytes)` is a nice one-liner for this. It also has `read_text()` for reading and `write_text()` for writing.